server/src/main/java/org/cloudfoundry/identity/uaa/provider/JdbcIdentityProviderProvisioning.java

```diff
@@ -56,6 +60,19 @@ public JdbcIdentityProviderProvisioning(JdbcTemplate jdbcTemplate) {
         this.jdbcTemplate = jdbcTemplate;
     }
 
+    public boolean idpWithAliasExistsInZone(final String zoneId) {
+        final Integer numberOfIdpsWithAlias = jdbcTemplate.queryForObject(
+                NUMBER_OF_IDPS_WITH_ALIAS_QUERY,
+                new Object[]{zoneId},
+                new int[]{VARCHAR},
+                Integer.class
+        );
+        if (numberOfIdpsWithAlias == null) {
+            throw new IncorrectResultSizeDataAccessException(1);
+        }
+        return numberOfIdpsWithAlias > 0;
+    }
+
```

Comment:
I tried to use `exists` in two different ways:

1. `VALUES (EXISTS (SELECT ...))` (see https://github.com/cloudfoundry/uaa/pull/2850/commits/e19a904e6cfa922537175df2128ac2be491ee8db#diff-8e7d64a1881ae78ed33d745ed1fd3a922312b89e740a3e1c75d1284b6a041b31R39) &rarr; supported by PostgreSQL and HSQL, but not MySQL
2. `SELECT EXISTS (SELECT ...)` &rarr; supported by PostgreSQL and MySQL, but not HSQL

Therefore, I decided to use the `count` function.

Reply:
Should I add a database migration that creates an index for the `alias_zid` column?

Reply:
> I tried to use `exists` in two different ways:
> ...
> Therefore, I decided to use the `count` function.

I see. HSQL is the hindrance here. In that case, how about using `limit` clause as follows?
`select 1 from identity_provider idp where idp.identity_zone_id = ? and idp.alias_zid is not null and idp.alias_zid <> '' limit 1`
That would return 1 if the record exists or null otherwise. It should be better than `select count(*)`.

Then you can change your Java code to return as follows (no need for additional check for null):
`return 1 == idpWithAliasExists;`